lib/ansible/plugins/lookup/template.py

```diff
@@ -67,6 +68,8 @@ def run(self, terms, variables, **kwargs):
         variable_start_string = kwargs.get('variable_start_string', None)
         variable_end_string = kwargs.get('variable_end_string', None)
 
+        old_vars = self._templar._available_variables
```

Comment:
I could swear i created a `get_variables` method to avoid using the internal dict directly, but cannot find it now ...

Reply:
I can add it in a separate PR.

Reply:
ah, was part of become plugins but others decided to drop that method ...

Reply:
FWIW I created https://github.com/ansible/ansible/pull/55435.